apps/backend/src/posts/posts.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+import { CreatePostDto } from './dto/create-post.dto';
+import { GetPostsDto } from './dto/get-posts.dto';
+import { UpdatePostDto } from './dto/update-post.dto';
+
+@Injectable()
+export class PostsService {
+  constructor(private readonly prisma: PrismaService) {}
+  create(createPostDto: CreatePostDto, user: User) {
+    return this.prisma.post.create({
+      data: {
+        ...createPostDto,
+        author: {
+          connect: {
+            authSchId: user.authSchId,
+          },
+        },
+      },
+    });
+  }
+
+  findAll(getPostsDto: GetPostsDto) {
+    const skip = getPostsDto.page * getPostsDto.page_size;
+    return this.prisma.post.findMany({
+      skip,
+      take: Number(getPostsDto.page_size),
+    });
+  }
+
+  findOne(id: number) {
+    return this.prisma.post.findUnique({
+      where: {
+        id,
+      },
+      include: {
+        author: true,
+      },
+    });
+  }
+
+  async update(id: number, updatePostDto: UpdatePostDto) {
+    return this.prisma.post.update({
+      where: {
+        id,
+      },
+      data: updatePostDto,
+    });
+  }
+
+  async remove(id: number) {
+    return this.prisma.post.delete({
+      where: {
+        id,
+      },
+    });
+  }
```

Comment:
prisma update and delete methods throw an exception if no record is found, we should catch so no 500 error is thrown